docker-compose.yaml

```diff
@@ -0,0 +1,78 @@
+name: photoview
+
+volumes:
+  mariadb:
+  postgres:
+
+services:
+  mysql:
+    image: mariadb:11.4
+    environment:
+      MYSQL_DATABASE: photoview_test
+      MYSQL_USER: photoview
+      MYSQL_PASSWORD: photosecret
+      MYSQL_RANDOM_ROOT_PASSWORD: yes
+    healthcheck:
+      test:
+      - CMD
+      - mariadb-admin
+      - --user=photoview
+      - --password=photosecret
+      - ping
+      interval: 10s
+      timeout: 5s
+      retries: 5
+    volumes:
+    - mariadb:/var/lib/mysql:rw
+    ports:
+    - 3306:3306
+
+  postgres:
+    image: postgres:16.3
+    environment:
+      POSTGRES_DB: photoview_test
+      POSTGRES_USER: photoview
+      POSTGRES_PASSWORD: photosecret
+    healthcheck:
+      test:
+      - CMD
+      - pg_isready
+      interval: 10s
+      timeout: 5s
+      retries: 5
+    volumes:
+    - postgres:/var/lib/postgresql/data:rw
+    ports:
+    - 5432:5432
+
+  api:
+    image: photoview/photoview-api
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: api
```

Comment:
Ah, now I see the purpose of copying all the repo inside the layer in Dockerfile.
Well, it will still cause bigger layers, bigger cache, and longer build of the PROD image on CI, but we can have the single Dockerfile, serving both workflows.
This approach might work, but as most hybrid solutions, it is not so flexible and has noticeable limitations for both use cases.

What if we create separate Dockerfiles inside `ui` and `api` folders? then we're free to implement there whatever is needed for the development use case without any impact on the PROD build workflow?

Reply:
I don't want to create a seperated Docker files, as we may sync them if the building workflow changes.

Could you give some examples about flexible and limitations for both use cases?

Reply:
I also hope to build with proper VCS info in go binary, https://pkg.go.dev/runtime/debug#BuildSetting. If we copy a part of repo in Dockerfile, either no `vcs` info because of no `.git`, or `vcs.modified` will be true, because the repo is not full in building env.

Reply:
As the file is updated, I resolve this thread. Please open a new one based on the latest file. Thank you.